[yu7ong] iP

yu7ong opened the pull request:
#DukePro 
> "Your mind is for having ideas, not holding them." - David Allen (source) 
DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yu7ong committed:
Rename, Greet, Exit

yu7ong committed:
Echo

yu7ong committed:
Add, List

yu7ong committed:
Mark as Done

yu7ong committed:
ToDos, Events, Deadlines

yu7ong committed:
Automated Text UI Testing

yu7ong committed:
Handle Errors

yu7ong committed:
Delete

yu7ong committed:
Add save the tasks in the hard disk

yu7ong committed:
Merge branch 'branch-Level-7'

yu7ong committed:
Add dates and times

yu7ong committed:
Merge branch 'branch-Level-8'

yu7ong committed:
Use more OOP

yu7ong committed:
Divide classes into packages

yu7ong committed:
Change tests

yu7ong committed:
Merge branch 'add-gradle-support'

yu7ong committed:
Add Junit tests for parsers

yu7ong committed:
Create a jar file

yu7ong committed:
Merge branch 'create-jar-file'

yu7ong committed:
Add JavaDoc

yu7ong committed:
Follow the Coding Standard

yu7ong committed:
Add find task function

yu7ong committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/parsers/Parser.java
#	src/main/java/duke/parsers/Time.java

yu7ong committed:
Merge branch 'branch-Level-9'

stanleytangzh commented on src/main/java/duke/parsers/Parser.java:
```diff
@@ -0,0 +1,154 @@
+        }
+        try {
+            if (mark) {
+                taskList.markTask(Integer.parseInt(split[1]));
```
i like how you separated markTask from unmarkTask.

stanleytangzh commented on src/main/java/duke/parsers/Time.java:
```diff
@@ -0,0 +1,53 @@
+
+    private LocalDate date;
+    private LocalDateTime dateTime;
+    private boolean hasTime;
```
This naming of boolean is good.

stanleytangzh commented on src/main/java/duke/tasks/TaskListStub.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    @Override
+    public void addTask(Task task) {
```
I like how you use verbs to name methods

stanleytangzh commented on src/main/java/duke/tasks/TaskListStub.java:
```diff
@@ -0,0 +1,30 @@
+
+public class TaskListStub extends TaskList {
+
+    List<Task> listOfTasks;
```
The use of plural is good.

stanleytangzh reviewed (commented):
LGTM.

yu7ong committed:
Correct coding style violations

natlv commented on src/main/java/duke/data/TaskDataBase.java:
```diff
@@ -0,0 +1,88 @@
+package duke.data;
+import java.io.*;
```
Perhaps consider only importing the specific package you need, instead of using *

natlv commented on src/main/java/duke/parsers/Parser.java:
```diff
@@ -0,0 +1,155 @@
+        taskList.addTask(new Todo(taskName));
+    }
+
+    private void handleAddDeadline(String message) throws MissingDateException,
```
I like how intuitive your function names are !

natlv reviewed (commented):
LGTM. Just some nits to fix

yu7ong committed:
Add GUI

yu7ong committed:
Merge branch 'branch-Level-10'

yu7ong committed:
Add assertions

Assertions added to class Tasklist and TasklistDatabase to catch
potential issues at runtime.

Adding checks can ensure that the file is created if it doesn't exist
when loading tasks, and assert that the task list is not null after
loading. Ensure tasks are correctly added and deleted, and that
marking and unmarking tasks work without accessing an invalid index.

This approach guarantees that operations on the task list are safe
and prevent potential runtime issues by asserting valid conditions.

yu7ong committed:
Improve code quality

Methods in Parser class have some common behaviors that cause
code duplication. Methods in Parser class use magic numbers.

Method in Time class has a nested try-catch block.

Removing magic numbers and replacing them with named constants and
extracting common logic into a single method allows us to improve
readability and reduce code duplication.

Extracting date and time parsing logic in Time class into a separate,
we can improve clarity.

This improves the maintainability and readability of the code.

yu7ong committed:
Improve code quality

yu7ong committed:
Merge pull request #1 from yu7ong/branch-A-Assertions

Add assertions

yu7ong committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/data/TaskDataBase.java
#	src/main/java/duke/tasks/TaskList.java

yu7ong committed:
Merge branch 'branch-A-CodeQuality'

yu7ong committed:
Add extension B-Reminder

yu7ong committed:
Merge branch 'master' into branch-A-CodingStandard

yu7ong committed:
Correct coding standard

yu7ong committed:
Improve the GUI

yu7ong committed:
Merge branch 'branch-A-BetterGUI'

yu7ong committed:
Add error handling for error in data file

yu7ong committed:
Merge branch 'branch-A-MoreErrorHandling'

yu7ong committed:
Fix unit tests

yu7ong committed:
Add Javadoc

yu7ong committed:
Fix bug in TaskDataBase

yu7ong committed:
Add User Guide

yu7ong committed:
Correct coding style violations
